dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/TrackerBundle.java

```diff
@@ -165,6 +167,24 @@ public Set<UID> getUpdatedTrackedEntities() {
     return Set.copyOf(this.updatedTrackedEntities);
   }
 
+  public List<SingleEvent> getSingleEvents() {
+    return this.events.stream()
+        .filter(event -> preheat.getProgram(event.getProgram()) != null)
+        .filter(event -> preheat.getProgram(event.getProgram()).isWithoutRegistration())
+        .map(event -> new SingleEvent())
+        .toList();
+  }
+
+  public List<TrackerEvent> getTrackerEvents() {
+    return this.events.stream()
+        .filter(
+            event ->
+                preheat.getProgram(event.getProgram()) == null
```

Comment:
why do we assume this is a tracker event if the program is not found?

Reply:
At this point I am still not sure when `getTrackerEvents` and `getSingleEvents` will be used and I want to be sure no event gets lost, so I needed to put somewhere the invalid events that have no program.